modules/server/src/test/scala/higherkindness/compendium/core/ProtocolUtilsStub.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2018-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.compendium.core
+
+import cats.effect.IO
+import higherkindness.compendium.models.Protocol
+
+class ProtocolUtilsStub(val pro: Protocol, val valid: Boolean) extends ProtocolUtils[IO] {
+  def validateProtocol(protocol: Protocol): IO[Protocol] =
+    if (valid) IO(pro)
```

Comment:
Can we use `mouse`'s fold here?

Reply:
Is mouse already in the project? Are we going to add another dep just for this?

Reply:
It's only being used here https://github.com/higherkindness/compendium/blob/f2be17ee2b679db23ecbe72c7be5c021311518e0/modules/server/src/main/scala/higherkindness/compendium/http/RootService.scala#L49

Now that you say it, it might be better to delete the mouse reference if we use it only for this.

Reply:
I like `Mouse` for cases as you put above or the suggested one is really useful, I would keep it and use it in obvious cases such as:
```scala
val a: Boolean = ...

a.fold(IO.pure(...), IO.raiseError(...))
``` 

Rather than
```
val a: Boolean = ...

if (a) IO.pure(...) else IO.raiseError(...)
```

But as you wish 🙂